plugins/s3_auth/s3_auth.cc

```diff
@@ -876,7 +953,13 @@ event_handler(TSCont cont, TSEvent event, void *edata)
   switch (event) {
   case TS_EVENT_HTTP_SEND_REQUEST_HDR:
     if (request.initialize()) {
-      status = request.authorize(s3);
+      while (true) {
```

Comment:
Why is this busy waiting desirable, especially since it seems to be redundant with the mutex.

Reply:
Because there's no guarantee there's no writer starvation with just the mutex.

Reply:
Ahh good point.  In my implementation of "one writer multi reader" locks, I made sure there would be no writer starvation:  https://github.com/ywkaras/trafficserver/blob/ecde23cf0368a47e4124612bbbbb75341d901463/src/tscore/OneWriterMultiReader.cc#L54 .  But, looking at cppreference.com, it doesn't say that std::shared_mutex provided a guarantee against writer starvation.  The PingPong approach would prevent writer starvation even if the std::shared_mutex implementation does not.

Reply:
https://stackoverflow.com/questions/57706952/does-stdshared-mutex-favor-writers-over-readers